django/thunderstore/frontend/extract_props.py

```diff
@@ -0,0 +1,34 @@
+import base64
+import json
+import re
+from io import StringIO
+from typing import Any, Iterable, Optional
+
+from lxml import etree
+
+REGEX_TEMPLATE = r"window\.ts\.{component}\(\s+document\.getElementById\(\"{id}\"\),\s+\"([a-zA-Z0-9=]+)\"\s+\);"
+
+
+def enumerate_scripts(html: str) -> Iterable[str]:
+    buffer = StringIO(html)
+    tree = etree.parse(buffer, etree.HTMLParser())
+    for entry in tree.findall('.//script[@type="text/javascript"]'):
```

Comment:
HTML spec recommends omitting the default type attribute from script elements. This kinda requires developers to know to not do that.

Reply:
I didn't intend this to be a perfect solution since it's only used for tests (in which case it failing is fine), but since it's in a module that doesn't indicate that I might as well make the fix to avoid misuse in the future.

Reply:
You have brought this rabbit hole upon yourself https://mathiasbynens.be/demo/javascript-mime-type